ws-client/src/client.rs

```diff
@@ -280,7 +305,7 @@ async fn background_task(
 	mut sender: jsonrpc_transport::Sender,
 	receiver: jsonrpc_transport::Receiver,
 	mut frontend: mpsc::Receiver<FrontToBack>,
-	config: WsConfig,
+	channel_capacities: ChannelCapacities,
```

Comment:
Maybe `ChannelCapacities` could be `ChannelConfig`? Or they could be in-lined directly here perhaps?

Reply:
Using `channel` in name is leaking abstraction, I reckon `concurrent|cy_limits` would be better here.

In general though, I don't see why this needs to be a separate struct at all, I'd just have separate fields.

Reply:
The intention was not to pass/clone to entire `Config` struct to the background thread when having `non-static` borrows, the background thread requires `static`.  However, we could just pass the `subscription capacity` and keep all fields in `Config` instead.

Reply:
>  I'd just have separate fields.

I like this. If cloning the config ever becomes a performance problem, maybe we have bigger problems?